### PR TITLE
Add missing links for v0.46 release and move v0.42 to EOL section

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

/kind documentation

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [ ] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [ ] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [ ] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Review by jerop — APPROVED** [review]
Thanks @AlanGreene

File: releases.md [hunk 1]
```diff
@@ -124,6 +124,7 @@ Older releases are EOL and available on [GitHub][tekton-pipeline-releases].
 [release-notes-standards]:
     https://github.com/tektoncd/community/blob/main/standards.md#release-notes
 
+[v0-46-0]: https://github.com/tektoncd/pipeline/releases/tag/v0.46.0
```
Comment (pritidesai) [inline releases.md:127]: oops, sorry for missing them @AlanGreene, thanks for fixing!

**Pull request merged**